src/appleseed.python/bindappleseedinfo.cpp

```diff
@@ -0,0 +1,99 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2018 Jonathan Dent, the appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// appleseed.foundation headers.
+#include "foundation/core/appleseed.h"
+#include "foundation/core/thirdparties.h"
+#include "foundation/platform/python.h"
+
+namespace bpy = boost::python;
+using namespace foundation;
+
+namespace
+{
+    // Assembles version information on all third party dependencies
+    bpy::dict get_deps_lib_versions()
+    {
+        bpy::dict output_dict;
+        const LibraryVersionArray& info_array = ThirdParties::get_versions();
```

Comment:
`ThirdParties::get_versions()` returns a value, not a reference, hence it's cleaner in my opinion to acknowledge that fact and make `info_array` a value too instead of a reference (even if in practices it doesn't change anything since the value referenced by `info_array` would live long enough for the reference not to dangle).